lib/notifications/bridge.ex

```diff
@@ -64,7 +71,6 @@ defmodule Notifications.Bridge do
     Process.send_after(pid, :update, @fetch_ms)
   end
 
-  @spec fetch_status() :: __MODULE__ | nil
```

Comment:
Even if we can no longer use `__MODULE__`, we should still have a typespec here.

Reply:
I removed these two existing typespecs because of a problem I ran into with Dialyzer. I forget the exact location, but somewhere there's code like:

```
# foo is either a Foo struct or nil
if(foo), do: process_thing(foo)

@spec process_thing(Foo.t()) :: Bar.t()
def process_thing(foo) do
  # ...do some stuff
end 
```

The problem is that `process_thing` is not speced to take a `nil` value, and unlike Typescript, Dialyzer doesn't detect that it's impossible for `process_thing` to get called with a `nil` value due to the `if` check--so you get a spurious error.

In general, I am very dubious about the value of typespecs and Dialyzer. I see it as a worst-of-both-worlds situation: you have all the extra formalities and boilerplate of type annotations, without the assurances an actual static type system gives you.

Reply:
Hmm, that's weird, I was just playing around with some examples and something similar to this passed dialyzer in my case:

```
  @spec foo(integer() | nil) :: integer()
  defp foo(i) do
    if i do
      bar(i)
    else
      0
    end
  end

  @spec bar(integer()) :: integer()
  defp bar(i) do
    i + 1
  end
```

FWIW I see the point of dialyzer as much more documentation and much less the sort of assurances you would associate with a static type system normally.